tools/system_libs.py

```diff
@@ -742,7 +742,7 @@ def vary_on(cls):
 
   @classmethod
   def get_default_variation(cls, **kwargs):
-    return super().get_default_variation(is_mt=settings.PTHREADS, is_ww=settings.WASM_WORKERS and not settings.PTHREADS, **kwargs)
+    return super().get_default_variation(is_mt=settings.PTHREADS, is_ww=settings.SHARED_MEMORY and not settings.PTHREADS, **kwargs)
```

Comment:
Is this needed in this PR?

Reply:
@eyebrowsonfire, is this maybe enough to resolve https://github.com/emscripten-core/emscripten/pull/22683

Reply:
@eyebrowsoffire, is this maybe enough to resolve #22683

Reply:
The reason this is needed for this PR is because it makes `__errno_location` into a function that, in non-threads build, has TLS erased, making it incompatible with `-sSHARED_MEMORY`, which triggers the problem found in #22683.   Basically it make the problem found in #22683 happen is way more cases, including simple hello world programs.